### PR TITLE
Verify a specific plugin version

### LINKED ISSUE #106: Add `--version` param to `wp plugin verify-checksums`
## Feature Request

- [x] Yes, I reviewed the [contribution guidelines](https://make.wordpress.org/cli/handbook/contributing/).

**Describe your use case and the problem you are facing**

For `wp core verify-checksums` the `--version` option is available to specify WP core version.
This is very useful in a deploy (update) process to check if the codebase is on the expected version.
If verify returns success, then skip download and/or report no change, if verification fails then download the right (usually newer) version. So it's possible to create an idempotent process.

It would be great to apply this logic to plugins also!

**Describe the solution you'd like**

Adding a `--version` option to `plugin verify-checksums` where we could directly set the version not just getting that from the plugin.
Of course, this would make sense only if a single plugin is provided, but this can be handled by ignoring `--version` if `--all` or multiple plugins are passed in.

Note: I'm happy to submit a PR.

### PR DESCRIPTION
**Pull request by semseysandor**
closes #106

### CODE REVIEW

**Comment by semseysandor** [comment]
Functional tests are failing, and I don't know the reason. I've tried a few variations and I think the code is working as expected.

Checking the output from "Run Behat" step:
- `wp plugin verify-checksums duplicate-post --format=json --version=3.2.2` returns with exit code 1
- stdout lists differences between `3.2.1` and `3.2.2`
- stderr has the error message

I believe the same assertions are specified in the scenario (or at least that was my intention), and this is what should happen.

@danielbachhuber do you have any idea?

File: features/checksum-plugin.feature [hunk 1]
```diff
@@ -13,6 +13,17 @@ Feature: Validate checksums for WordPress plugins
       Success: Verified 1 of 1 plugins.
       """
 
+    When I run `wp plugin verify-checksums duplicate-post --format=json --version=3.2.2`
```
Comment (wojsmol) [inline features/checksum-plugin.feature:16]:
```suggestion
    When I try `wp plugin verify-checksums duplicate-post --format=json --version=3.2.2`
```
In Bahat we use `When I try` when error is expected.

Comment (semseysandor) [reply features/checksum-plugin.feature]: cool, thanks!

**Comment by semseysandor** [comment]
OK, now tests are passing, it's ready for review!

**Review by danielbachhuber — COMMENTED** [review]
Looks pretty good! Just a few nits to address, and then it should be ready to ship.

File: src/Checksum_Plugin_Command.php [hunk 2]
```diff
@@ -88,7 +92,7 @@ public function __invoke( $args, $assoc_args ) {
 		$skips = 0;
 
 		foreach ( $plugins as $plugin ) {
-			$version = $this->get_plugin_version( $plugin->file );
+			$version = empty( $version_cli ) ? $this->get_plugin_version( $plugin->file ) : $version_cli;
```
[the remark this answers is not in the document]
Comment (danielbachhuber) [reply src/Checksum_Plugin_Command.php:95]: Ah, sorry, missed that. Maybe `$version_arg` to distinguish, then?

File: src/Checksum_Plugin_Command.php [hunk 3]
```diff
@@ -75,11 +78,12 @@ class Checksum_Plugin_Command extends Checksum_Base_Command {
...
-		$all      = (bool) Utils\get_flag_value( $assoc_args, 'all', false );
-		$strict   = (bool) Utils\get_flag_value( $assoc_args, 'strict', false );
-		$insecure = (bool) Utils\get_flag_value( $assoc_args, 'insecure', false );
-		$plugins  = $fetcher->get_many( $all ? $this->get_all_plugin_names() : $args );
+		$fetcher     = new Fetchers\UnfilteredPlugin();
+		$all         = (bool) Utils\get_flag_value( $assoc_args, 'all', false );
+		$strict      = (bool) Utils\get_flag_value( $assoc_args, 'strict', false );
+		$insecure    = (bool) Utils\get_flag_value( $assoc_args, 'insecure', false );
+		$plugins     = $fetcher->get_many( $all ? $this->get_all_plugin_names() : $args );
+		$version_cli = isset( $assoc_args['version'] ) ? $assoc_args['version'] : '';
```
[the remark this answers is not in the document]
Comment (semseysandor) [reply src/Checksum_Plugin_Command.php:86]: I think the minimalistic solution to elimate the need for a new local var is to inline this in:
```
$version = isset( $assoc_args['version'] ) ? $assoc_args['version'] : $this->get_plugin_version( $plugin->file );
```
I chose this more verbose way to keep variable initialization in one block, but I've got no problem with the inlined version also.

File: src/Checksum_Plugin_Command.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (semseysandor) [reply src/Checksum_Plugin_Command.php]: OK

**Review by danielbachhuber — APPROVED** [review]
Thanks for your work on this, @semseysandor !

**Pull request merged**